Modules/FindESMF.cmake

```diff
@@ -79,11 +79,23 @@ if (ESMF_FOUND)
   endforeach()
   set(ESMF_F90COMPILEPATHS ${tmp})
 
-  add_library(esmf SHARED IMPORTED)
+  add_library(esmf UNKNOWN IMPORTED)
   find_library(esmf_lib NAMES esmf_fullylinked PATHS ${ESMF_LIBSDIR})
+  if(esmf_lib MATCHES "esmf_lib-NOTFOUND")
+    message(STATUS "Dynamic/fully-linked ESMF library not found, searching for static library instead")
+    find_library(esmf_lib NAMES esmf PATHS ${ESMF_LIBSDIR})
+    if(esmf_lib MATCHES "esmf_lib-NOTFOUND")
+      message(FATAL_ERROR "Neither dynamic nor static ESMF library were found")
```

Comment:
I need to change this to a warning message, the FindESMF package should not abort the cmake call.